### PR TITLE
Repurpose MinionInstancesCleanupTask to StaleInstancesCleanupTask to remove stale broker/server instances

### PR DESCRIPTION
**Pull request by xiangfu0**
Repurpose the periodic controller task MinionInstancesCleanupTask to StaleInstancesCleanupTask to allow remove stale broker/server instances as well.

Adding new configs for the periodical task:
```
controller.stale.instances.cleanup.task.initialDelaySeconds, default to random 2-5min
controller.stale.instances.cleanup.task.frequencyPeriod, default to 1h
controller.stale.instances.cleanup.task.minOfflineTimeBeforeDeletionPeriod, default to 1h
```

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/cleanup/StaleInstancesCleanupTask.java [hunk 1]
```diff
@@ -82,7 +95,47 @@ protected void runTask(Properties periodicTaskProperties) {
...
+          PinotResourceManagerResponse response = _pinotHelixResourceManager.dropInstance(offlineInstance);
+          if (response.isSuccessful()) {
+            _controllerMetrics.addValueToGlobalGauge(ControllerGauge.DROPPED_SERVER_INSTANCES, 1);
+          }
+        }
       }
     }
   }
+
+  private Set<String> getBrokerInstancesInUse() {
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/cleanup/StaleInstancesCleanupTask.java:128]: No need to read all tables. Only need to read the `brokerResource` IS.

**Review by Jackie-Jiang — COMMENTED** [review]
I don't think we need to keep both minion only mode and all instances mode. We can deprecate the minion ones

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/cleanup/StaleInstancesCleanupTask.java [hunk 2]
```diff
@@ -34,23 +36,30 @@
...
+public class StaleInstancesCleanupTask extends BasePeriodicTask {
+  private static final Logger LOGGER = LoggerFactory.getLogger(StaleInstancesCleanupTask.class);
+  private final static String TASK_NAME = "StaleInstancesCleanupTask";
+
   protected final PinotHelixResourceManager _pinotHelixResourceManager;
   protected final LeadControllerManager _leadControllerManager;
   protected final ControllerMetrics _controllerMetrics;
+  // This applies to both broker and server instances.
+  private final long _staleInstancesCleanupTaskMinOfflineTimeBeforeDeletionInMilliseconds;
   private final long _minionInstanceCleanupTaskMinOfflineTimeBeforeDeletionInMilliseconds;
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/cleanup/StaleInstancesCleanupTask.java]: removed.

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/cleanup/StaleInstancesCleanupTask.java (hunk 1 above)
Comment (xiangfu0) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/cleanup/StaleInstancesCleanupTask.java:129]: updated to use brokerResource

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 3]
```diff
@@ -139,6 +139,14 @@ public static class ControllerPeriodicTasksConf {
         "controller.minion.instances.cleanup.task.minOfflineTimeBeforeDeletionSeconds";
     public static final String MINION_INSTANCES_CLEANUP_TASK_MIN_OFFLINE_TIME_BEFORE_DELETION_PERIOD =
         "controller.minion.instances.cleanup.task.minOfflineTimeBeforeDeletionPeriod";
+
+    public static final String STALE_INSTANCES_CLEANUP_TASK_FREQUENCY_PERIOD =
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java:146]: done

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java [hunk 4]
```diff
@@ -99,9 +99,16 @@ public enum ControllerGauge implements AbstractMetrics.Gauge {
...
+  // Number of dropped stale minion instances
   DROPPED_MINION_INSTANCES("droppedMinionInstances", true),
 
+  // Number of dropped stale broker instances
+  DROPPED_BROKER_INSTANCES("droppedBrokerInstances", true),
+
+  // Number of dropped stale server instances
+  DROPPED_SERVER_INSTANCES("droppedServerInstances", true),
+
+
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-common/src/main/java/org/apache/pinot/common/metrics/ControllerGauge.java]: done

**Review by Jackie-Jiang — APPROVED** [review]

**Comment by Jackie-Jiang** [comment]
Please update the PR description with some release note, and also update the pinot doc about the new configs

**Comment by xiangfu0** [comment]
Updated to https://docs.pinot.apache.org/basics/components/controller#staleinstancescleanuptask

**Pull request merged**